ccmlib/node.py

```diff
@@ -565,14 +565,26 @@ def watch_log_for(self, exprs, from_mark=None, timeout=600, process=None, verbos
                     if start + timeout < time.time():
                         raise TimeoutError(time.strftime("%d %b %Y %H:%M:%S", time.gmtime()) + " [" + self.name + "] Missing: " + str([e.pattern for e in tofind]) + ":\n" + reads[:50] + ".....\nSee {} for remainder".format(filename))
 
-                if process:
-                    if common.is_win():
-                        if not self.is_running():
-                            return None
-                    else:
-                        process.poll()
-                        if process.returncode == 0:
-                            return None
+                    # update status because meanwhile node could have died....
+                    # if it was running but now it is not we should return immediately rather than after timeout
+                    if self.pid and not self._is_pid_running():
+                        msg = "node {n} with pid {pid} stopped during watching log for {m}".\
+                            format(n=self.name, pid=self.pid, m=exprs)
+                        common.warning(msg)
+                        raise NodeError(msg)
+
+                # this can preliminary end checking (if child process is terminated)
```

Comment:
IIUC: if child process terminated => return None
but this None is not checked anywhere....